tools/codegen/api/cpp.py

```diff
@@ -191,9 +195,89 @@ def argument(a: Union[Argument, TensorOptionsArguments, ThisArgument]) -> CppArg
     else:
         assert_never(a)
 
-def group_arguments(
-    func: FunctionSchema, *, method: bool = False
-) -> Sequence[Union[Argument, TensorOptionsArguments, ThisArgument]]:
+@dataclass(frozen=True)
+class CppSignature:
+    arguments: Sequence[Union[Argument, TensorOptionsArguments, ThisArgument]]
```

Comment:
Make this Tuple instead of Sequence, then this dataclass will be hashable.

Reply:
A tuple has a fixed number of elements right? How does this work for a variable length argument list?

Reply:
`Tuple[Union[Argument, TensorOptionsArguments, ThisArgument], ...]`